server/v2/store/server.go

```diff
@@ -0,0 +1,59 @@
+package store
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"cosmossdk.io/core/transaction"
+	"cosmossdk.io/log"
+	serverv2 "cosmossdk.io/server/v2"
+)
+
+type StoreComponent struct {
+	config *Config
+}
+
+func New() serverv2.ServerComponent[transaction.Tx] {
+	return StoreComponent{}
+}
+
+func (s StoreComponent) Init(appI serverv2.AppI[transaction.Tx], v *viper.Viper, logger log.Logger) (serverv2.ServerComponent[transaction.Tx], error) {
```

Comment:
should it be a pointer? so we don't need to use the result of init.

Reply:
done here: https://github.com/cosmos/cosmos-sdk/pull/20742

Reply:
let's merge it first an then I'll sync from main